### PR TITLE
fix: max window size defaults to 0

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/32995.

Fixes an issue where setting window `maxHeight` or `maxWidth` made it so the width and height could no longer be resized.

https://github.com/electron/electron/pull/32628 fixed an issue with `maxWidth` not being respected at all, but also changed defaults such that `max_width` and `max_height` conformed to `size_constraints.GetMaximumSize()`. By default, `size_constraints.GetMaximumSize()` returns `(0, 0)` unless it's been previously set, so this accidentally made it such that the `maxHeight` couldn't be changed if the `maxWidth` was set, and vice versa. This fixes that issue.
#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an issue where setting window `maxHeight` or `maxWidth` made it so the width and height could no longer be resized.

### CODE REVIEW

File: patches/chromium/fix_don_t_restore_maximized_windows_when_calling_showinactive.patch [hunk 1]
```diff
@@ -7,10 +7,10 @@ This is a backport from Chromium of
 https://chromium-review.googlesource.com/c/chromium/src/+/3371573.
 
 diff --git a/ui/views/win/hwnd_message_handler.cc b/ui/views/win/hwnd_message_handler.cc
-index 067861bb743ee2f3c1916794d45efb7dd591b230..6507557bf5a47492343602607e0dbb7d8f88246d 100644
+index 4be0e5b0ef187d68f054672bc3d1dc328057c58b..264a9109e42c23e9be6bf7269b3cfee2634b61e4 100644
```
Comment (nornagon) [inline patches/chromium/fix_don_t_restore_maximized_windows_when_calling_showinactive.patch:10]: what's up with this patch change? seems unrelated.

**Review by nornagon — APPROVED** [review]

File: patches/chromium/fix_don_t_restore_maximized_windows_when_calling_showinactive.patch (hunk 1 above)
Comment (codebytere) [reply patches/chromium/fix_don_t_restore_maximized_windows_when_calling_showinactive.patch:10]: whoops guess that snuck in there from something else i was working on - removing!

**Review by zcbenz — CHANGES_REQUESTED** [review]
Instead of setting maximum size to `(INT_MAX, INT_MAX)`, the code should only call `size_constraints.set_maximum_size` when there is a maximum size set.

By default the window has a default maximum size that prevents it from being resized larger than screen, assigning a maximum size would override this behavior, see also:

https://github.com/electron/electron/blob/bcf060fab67fe1912d03ad253de98b3091b2a61b/shell/browser/native_window_views.cc#L211-L217

File: shell/browser/native_window.cc [hunk 2]
```diff
@@ -161,16 +161,25 @@ void NativeWindow::InitFromOptions(const gin_helper::Dictionary& options) {
...
+  gfx::Size max_size = size_constraints.GetMaximumSize();
+  int max_width = max_size.width() > 0 ? max_size.width() : INT_MAX;
+  int max_height = max_size.height() > 0 ? max_size.height() : INT_MAX;
+  bool have_max_width = options.Get(options::kMaxWidth, &max_width);
+  bool have_max_height = options.Get(options::kMaxHeight, &max_height);
+
+  // By default the window has a default maximum size that prevents it
+  // from being resized larger than the screen, so we should only set this
+  // if th user has passed in values.
+  if (have_max_height || have_max_width)
```
Comment (zcbenz) [inline shell/browser/native_window.cc:180]: Should also check if `max_size` is not empty.

Comment (codebytere) [reply shell/browser/native_window.cc:180]: done!

**Review by zcbenz — APPROVED** [review]

**Comment by codebytere** [comment]
Failure not relevant - merging.
